src/data/polynomial/laurent.lean

```diff
@@ -0,0 +1,331 @@
+non-positive degree of `f` are summed up and used as constant term. -/
+def trunc : R[T;T⁻¹] →+ R[X] :=
+((to_finsupp_iso R).symm.to_add_monoid_hom).comp (map_domain.add_monoid_hom int.to_nat)
+
+@[simp]
+lemma trunc_single_nat (n : ℕ) (r : R) : trunc (C r * T n) = monomial n r :=
+begin
+  apply (to_finsupp_iso R).injective,
+  -- after `rw ← single_eq_C_mul_T`, the combination `simp [-single_eq_C_mul_T, trunc], refl`
+  -- closes the goal, but involves a non-terminal `simp`.
+  rw [← single_eq_C_mul_T, trunc, add_monoid_hom.coe_comp, function.comp_app,
+    map_domain.add_monoid_hom_apply, map_domain_single, int.to_nat_coe_nat, to_finsupp_iso_apply,
+    to_finsupp_iso_apply, to_finsupp_monomial],
+  refl,
+end
+
+@[simp]
+lemma trunc_single_neg_nat (n : ℕ) (r : R) : trunc (C r * T (- n)) = polynomial.C r :=
+begin
+  apply (to_finsupp_iso R).injective,
+  -- after `rw ← single_eq_C_mul_T`, the combination `simp [-single_eq_C_mul_T, trunc], refl`
+  -- closes the goal, but involves a non-terminal `simp`.
+  rw [← single_eq_C_mul_T, trunc, add_monoid_hom.coe_comp, function.comp_app,
+    map_domain.add_monoid_hom_apply, to_finsupp_iso_apply, map_domain_single, int.to_nat_neg_nat,
+    to_finsupp_iso_apply, to_finsupp_C],
+  refl,
+end
+
+lemma polynomial.to_laurent_polynomial_injective :
+  function.injective (polynomial.to_laurent_polynomial : R[X] → R[T;T⁻¹]) :=
+begin
+  refine function.has_left_inverse.injective ⟨trunc, λ f, f.induction_on' _ _⟩,
+  { exact λ f g hf hg, by simp only [hf, hg, _root_.map_add] },
+  { exact λ n r, by simp only [monomial_eq_C_mul_T, trunc_single_nat] }
+end
+
+lemma exists_X_pow (f : R[T;T⁻¹]) : ∃ (n : ℕ) (f' : R[X]), f'.to_laurent_polynomial = f * T n :=
+begin
+  apply f.induction_on' _ (λ n a, _); clear f,
+  { rintros f g ⟨m, fn, hf⟩ ⟨n, gn, hg⟩,
+    by_cases h : m ≤ n;
+    refine ⟨m + n, fn * X ^ n + gn * X ^ m, _⟩;
+    simp only [hf, hg, add_mul, add_comm (n : ℤ), int.coe_nat_add, _root_.map_add,
+      _root_.map_mul, map_pow, X_eq_T, T_pow, mul_one, mul_T_assoc]  },
+  { cases n with n n,
+    { exact ⟨0, polynomial.C a * X ^ n, by simp⟩ },
+    { refine ⟨n + 1, polynomial.C a, _⟩,
+      simp only [int.neg_succ_of_nat_eq, neg_add_self, int.coe_nat_succ, mul_T_assoc, T_zero,
+        mul_one, C_eq_C] } }
+end
+
+lemma exists_X_pow_T' (f : R[T;T⁻¹]) :
+  ∃ (n : ℕ) (f' : R[X]), f'.to_laurent_polynomial = f * T' ^ n :=
+begin
+  rcases exists_X_pow f with ⟨n, f', hf⟩,
+  refine ⟨n, f', _⟩,
+  rwa [T', ← T, T_pow, mul_one],
+end
+
+/--  This version of `exists_X_pow` can be called as `rcases f.exists_X_pow' with ⟨n, f', rfl⟩`. -/
+lemma exists_X_pow' (f : R[T;T⁻¹]) : ∃ (n : ℤ) (f' : R[X]),
+  f = f'.to_laurent_polynomial * T n :=
+begin
+  rcases f.exists_X_pow with ⟨n, f', hf⟩,
+  exact ⟨(- n), f', by simp [hf]⟩,
+end
+
+/--  Suppose that `Q` is a statement about Laurent polynomials such that
+* `Q` is true on ~~Laurent~~ polynomials;
```

Comment:
```suggestion
* `Q` is true on ~~ordinary~~ polynomials;
```
I guess?

Reply:
Thanks!  I used your suggestion, and placed it in italic, rather than strikethrough.

Reply:
Oh ahah i understand now! But yes not all of us are used to ~~ indicating strike through i guess!